conans/client/profile_loader.py

```diff
@@ -59,17 +59,57 @@ def get_default_build(self):
     def _load_profile_plugin(self):
         profile_plugin = os.path.join(self._cache.plugins_path, "profile.py")
         if not os.path.isfile(profile_plugin):
-            profile_process = textwrap.dedent("""
-                def profile_plugin(profile):
-                    settings = profile.settings
-                    if settings.get("compiler") == "msvc" and settings.get("compiler.runtime"):
-                        if settings.get("compiler.runtime_type") is None:
-                            runtime = "Debug" if settings.get("build_type") == "Debug" else "Release"
-                            try:
-                                settings["compiler.runtime_type"] = runtime
-                            except ConanException:
-                                pass
-                """)
+            profile_process = """
+def profile_plugin(profile):
+    settings = profile.settings
+    if settings.get("compiler") == "msvc" and settings.get("compiler.runtime"):
+        if settings.get("compiler.runtime_type") is None:
+            runtime = "Debug" if settings.get("build_type") == "Debug" else "Release"
+            try:
+                settings["compiler.runtime_type"] = runtime
+            except ConanException:
+                pass
+    _check_correct_cppstd(settings)
+
+def _check_correct_cppstd(settings):
+    from conan.tools.scm import Version
+    def _error():
+        from conan.errors import ConanInvalidConfiguration
+        raise ConanInvalidConfiguration("The provided compiler.cppstd is not supported with the specified compiler")
```

Comment:
I think this can be a ConanException. We don't want to check compatibility, if a profile is broken/badly defined, it should error, report it as a broken profile and stop?

Reply:
Yes, I had the same doubt. Look like look are losing some potentially valuable information in the return code but we can improve later if needed. Let's raise ConanException